File: Programing_Basics_with_JS/conditionalStatementsAdvancedMoreExercises/06.TruckDriver.js
function truckDriver(input) {
    let season = input[0];
    let kilometersPerMonth = Number(input[1]);

    let totalKilometers = kilometersPerMonth * 4;
    let money = 0;

    if (kilometersPerMonth <= 5000) {
        switch(season) {
            case 'Summer': money = totalKilometers * 0.90; break;
            case 'Winter': money = totalKilometers * 1.05; break;
            case 'Spring':
            case 'Autumn': money = totalKilometers * 0.75; break;
        }
    } else if (kilometersPerMonth <= 10000) {
        switch(season) {
            case 'Summer': money = totalKilometers * 1.10; break;
            case 'Winter': money = totalKilometers * 1.25; break;
            case 'Spring':
            case 'Autumn': money = totalKilometers * 0.95; break;
        }
    } else if (kilometersPerMonth <= 20000) {
        switch(season) {
            case 'Summer': 
            case 'Winter': 
            case 'Spring':
            case 'Autumn': money = totalKilometers * 1.45; break;
        }
    }

    let finalMoney = money * 0.90;

    console.log(finalMoney.toFixed(2));
}

truckDriver(['Summer', '3455'])
truckDriver(['Winter', '4350'])
truckDriver(['Spring', '1600'])
truckDriver(['Winter', '5678'])
truckDriver(['Autumn', '8600'])
truckDriver(['Winter', '16042'])
truckDriver(['Spring', '16942'])

File: Programming_Basics_with_Python/conditional_statements_advanced_more_exercises/06._truck_driver.py
season = input()
km_per_month = float(input())

season_months = 4
profit_per_month = 0

if km_per_month <= 5000:
    if season == 'Spring' or season == 'Autumn':
        profit_per_month = km_per_month * 0.75
    elif season == 'Summer':
        profit_per_month = km_per_month * 0.90
    elif season == 'Winter':
        profit_per_month = km_per_month * 1.05

elif 5000 < km_per_month <= 10000:
    if season == 'Spring' or season == 'Autumn':
        profit_per_month = km_per_month * 0.95
    elif season == 'Summer':
        profit_per_month = km_per_month * 1.10
    elif season == 'Winter':
        profit_per_month = km_per_month * 1.25

elif 10000 < km_per_month <= 20000:
    profit_per_month = km_per_month * 1.45

final_profit = (profit_per_month * season_months) * 0.9

print(f'{final_profit:.2f}')